### DISCUSSION TITLE
Table Component Sort -- Ordering Contents Wrong

### DISCUSSION THREAD

**Discussion by BeckerIndustries** [olton/Metro-UI-CSS#1751] [Q&A] [answered]
Hi all,

I am working on a table that has a single-column sort.  However, it does not sort the contents properly.  I can click the header to re-sort the contents but they are still sorted out of order.

```
                <table class="table striped table-border mt-4" data-role="table" data-cell-wrapper="false" data-pagination="true">
                  <thead>
                    <tr>
                      <th></th>                    
                      <th class="sortable-column sort-desc" data-sortable="true" data-sort-dir="desc">System Name</th>
                      <th>Description</th>
                    </tr>
                  </thead>
                  <tbody>
                    <?php
                    $sql = "SELECT * FROM `phpbbom_links`";
                    $result = $db->sql_query($sql);
                    while($row = $db->sql_fetchrow($result)){ ?>
                        <tr>
                          <td><?php if($row['restricted'] == 1){ ?><span class="mif-security" title="Restricted"></span><?php } ?></td>
                          <td><span style="white-space: nowrap"><a href="redirect.php?id=<?php echo $row['id']; ?>" target="_blank"><?php echo $row['title']; ?></a></span></td>
                          <td style="word-wrap: break-word !important"><?php echo $row['description']; ?></td>
                        </tr>
                    <?php
                    }
                    $db->sql_freeresult($result); 
                    ?>
                  </tbody>
                </table>
```

**Comment by BeckerIndustries** [comment]
I am guessing because the data is loaded after the column, its not recognizing the data until its too late.  My solution is to SQL request the data in alpha order before populating the table so that it matches the tables expected sort order.

**Reply by BeckerIndustries** [reply]
Same results....   ```SELECT * FROM `phpbbom_links` ORDER BY `phpbbom_links`.`title` ASC``` through phpmyadmin sorts them in order correctly.... but on the page, the table still sorts them out of order.

**Comment by olton** [comment]
The component sorts table on the second field with a full cell value `<a ...>...</a>`, if you need a sort by anchor value, without a tag definition symbols, you must put into the cell a pure value, and with event `onDrawCell` transform its value into a link. In this case, a sort will work right.

Examples:
https://codepen.io/olton/pen/mdRwqBB
https://codepen.io/olton/pen/xxgrPWV?editors=1010

**Comment by olton** [comment, marked as answer]
Also component removes any inline styles, classes, and other attributes for a table cell (`td`). If you need to define additional styles for a table cell, you can:
1) Use special attributes for table component for adding classes - `data-cls-body-cell`, `data-cls-cell-wrapper`
2) Use event `onDrawCell` to rewrite the same table cell as you need
3) Create a style for a table cell over table
```css
    .table tr td {
         ...
    }
```

**Reply by BeckerIndustries** [reply]
But I need to link to a $row['url'] not the value of the cell contents.

**Reply by BeckerIndustries** [reply]
https://codepen.io/codymatthew/pen/rNjzboW?editors=1010

So this will get the value of the 3rd column and append to the hyperlink. but i dont want to show the 3rd column. i dont want the ID visible.

**Reply by olton** [reply]
https://codepen.io/olton/pen/xxgrPWV?editors=1010
